arkd: calling GetBalance when initialized=false panics

```sh
make run-signet
Cleaning...
Running arkd in signet mode...
INFO[0000] All 0 tables opened in 0s
INFO[0000] Discard stats nextEmptySlot: 0
INFO[0000] Set nextTxnTs to 0
INFO[0000] Deleting empty file: /Users/tiero/Library/Application Support/Arkd/db/round-events/000003.vlog
INFO[0000] 2024-08-14 16:55:31.577 [INF] neutrino: Loaded 0 addresses from file '/Users/tiero/Library/Application Support/Arkd/db/peers.json'
INFO[0000] 2024-08-14 16:55:31.578 [INF] neutrino: Registering block subscription: id=1
INFO[0000] 2024-08-14 16:55:31.578 [INF] neutrino: Broadcaster now active
INFO[0000] starting service...
INFO[0000] started listening at :7070
INFO[0000] 2024-08-14 16:55:31.580 [INF] neutrino: DNS discovery failed on seed x49.bitcoind-mutinynet.bitcoind.svc.cluster.local:38333: lookup x49.bitcoind-mutinynet.bitcoind.svc.cluster.local:38333: no such host
^CINFO[0227] shutting down service...
INFO[0227] stopped grpc server
Error: Logrus exit handler error: runtime error: invalid memory address or nil pointer dereference
make: *** [run-signet] Error 1
make run-signet
Cleaning...
Running arkd in signet mode...
INFO[0000] All 0 tables opened in 0s
INFO[0000] Discard stats nextEmptySlot: 0
INFO[0000] Set nextTxnTs to 0
INFO[0000] Deleting empty file: /Users/tiero/Library/Application Support/Arkd/db/round-events/000004.vlog
INFO[0000] 2024-08-14 17:05:27.538 [INF] neutrino: Loaded 0 addresses from file '/Users/tiero/Library/Application Support/Arkd/db/peers.json'
INFO[0000] 2024-08-14 17:05:27.539 [INF] neutrino: Registering block subscription: id=1
INFO[0000] 2024-08-14 17:05:27.539 [INF] neutrino: Broadcaster now active
INFO[0000] starting service...
INFO[0000] started listening at :7070
DEBU[0029] gRPC method: /ark.v1.WalletInitializerService/GetStatus
DEBU[0040] gRPC method: /ark.v1.WalletService/GetBalance
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x105a30f80]

goroutine 232 [running]:
github.com/lightningnetwork/lnd/lnwallet/btcwallet.(*BtcWallet).ListUnspentWitness(0x1400b4a1788?, 0x5178240?, 0x1?, {0x105be0a52?, 0x2c?})
	/Users/tiero/go/pkg/mod/github.com/lightningnetwork/lnd@v0.18.2-beta/lnwallet/btcwallet/btcwallet.go:1132 +0x20
github.com/lightningnetwork/lnd/lnwallet/btcwallet.(*BtcWallet).ConfirmedBalance(0x1400b4a1718?, 0x4d686a0?, {0x105be0a52?, 0x14000801830?})
	/Users/tiero/go/pkg/mod/github.com/lightningnetwork/lnd@v0.18.2-beta/lnwallet/btcwallet/btcwallet.go:480 +0x2c
github.com/ark-network/ark/server/internal/infrastructure/wallet/btc-embedded.(*service).getBalance(...)
	/Users/tiero/projects/ark-network/ark/server/internal/infrastructure/wallet/btc-embedded/wallet.go:845
github.com/ark-network/ark/server/internal/infrastructure/wallet/btc-embedded.(*service).MainAccountBalance(0x1400b4a1718?, {0x104d68178?, 0x105177c60?})
	/Users/tiero/projects/ark-network/ark/server/internal/infrastructure/wallet/btc-embedded/wallet.go:298 +0x30
github.com/ark-network/ark/server/internal/interface/grpc/handlers.(*walletHandler).GetBalance(0x14000286190, {0x106277dc8, 0x140008017a0}, 0x104de5b30?)
	/Users/tiero/projects/ark-network/ark/server/internal/interface/grpc/handlers/walletservice.go:125 +0x3c
github.com/ark-network/ark/api-spec/protobuf/gen/ark/v1._WalletService_GetBalance_Handler.func1({0x106277dc8?, 0x140008017a0?}, {0x1060fd500?, 0x140008017d0?})
	/Users/tiero/projects/ark-network/ark/api-spec/protobuf/gen/ark/v1/wallet_grpc.pb.go:372 +0xd0
github.com/ark-network/ark/server/internal/interface/grpc/interceptors.UnaryInterceptor.unaryMacaroonAuthHandler.func1({0x106277dc8, 0x140008017a0}, {0x1060fd500, 0x140008017d0}, 0x1?, 0x14000f3faa0)
	/Users/tiero/projects/ark-network/ark/server/internal/interface/grpc/interceptors/auth.go:25 +0x84
github.com/ark-network/ark/server/internal/interface/grpc/interceptors.UnaryInterceptor.ChainUnaryServer.func3.1({0x106277dc8?, 0x140008017a0?}, {0x1060fd500?, 0x140008017d0?})
	/Users/tiero/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/chain.go:48 +0x50
github.com/ark-network/ark/server/internal/interface/grpc/interceptors.unaryLogger({0x106277dc8, 0x140008017a0}, {0x1060fd500, 0x140008017d0}, 0x0?, 0x140005fb3c0)
	/Users/tiero/projects/ark-network/ark/server/internal/interface/grpc/interceptors/logger.go:17 +0x9c
github.com/ark-network/ark/server/internal/interface/grpc/interceptors.UnaryInterceptor.ChainUnaryServer.func3({0x106277dc8, 0x140008017a0}, {0x1060fd500, 0x140008017d0}, 0x14000527240, 0x106201dc0?)
	/Users/tiero/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.4.0/chain.go:53 +0x114
github.com/ark-network/ark/api-spec/protobuf/gen/ark/v1._WalletService_GetBalance_Handler({0x1060a58c0, 0x14000286190}, {0x106277dc8, 0x140008017a0}, 0x14000816700, 0x140008001b0)
	/Users/tiero/projects/ark-network/ark/api-spec/protobuf/gen/ark/v1/wallet_grpc.pb.go:374 +0x148
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002b3000, {0x106277dc8, 0x14000822d80}, {0x1062828a0, 0x1400112af70}, 0x14000ca87e0, 0x140008003c0, 0x106d70590, 0x0)
	/Users/tiero/go/pkg/mod/google.golang.org/grpc@v1.65.0/server.go:1379 +0xb58
google.golang.org/grpc.(*Server).handleStream(0x140002b3000, {0x1062828a0, 0x1400112af70}, 0x14000ca87e0)
	/Users/tiero/go/pkg/mod/google.golang.org/grpc@v1.65.0/server.go:1790 +0xb20
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	/Users/tiero/go/pkg/mod/google.golang.org/grpc@v1.65.0/server.go:1029 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 190
	/Users/tiero/go/pkg/mod/google.golang.org/grpc@v1.65.0/server.go:1040 +0x13c
exit status 2
make: *** [run-signet] Error 1
```

```sh
arkd --no-macaroon wallet balance
FATA[0000] Get "http://localhost:7070/v1/admin/wallet/balance": EOF
``

--- a/server/internal/infrastructure/wallet/btc-embedded/wallet.go
+++ b/server/internal/infrastructure/wallet/btc-embedded/wallet.go
@@ -221,8 +221,10 @@ func NewService(cfg WalletConfig, options ...WalletOption) (ports.WalletService,
 }
 
 func (s *service) Close() {
-	if err := s.wallet.Stop(); err != nil {
-		log.WithError(err).Warn("failed to gracefully stop the wallet, forcing shutdown")
+	if s.wallet != nil {
+		if err := s.wallet.Stop(); err != nil {
+			log.WithError(err).Warn("failed to gracefully stop the wallet, forcing shutdown")
+		}
 	}
 }
 
@@ -323,6 +325,10 @@ func (s *service) Unlock(_ context.Context, password string) error {
 }
 
 func (s *service) Lock(_ context.Context, _ string) error {
+	if err := s.expectWallet(); err != nil {
+		return err
+	}
+
 	s.wallet.InternalWallet().Lock()
 	return nil
 }
@@ -900,6 +906,10 @@ func (s *service) initWallet(wallet *btcwallet.BtcWallet) error {
 }
 
 func (s *service) getBalance(account accountName) (uint64, error) {
+	if err := s.expectWallet(); err != nil {
+		return 0, err
+	}
+
 	balance, err := s.wallet.ConfirmedBalance(0, string(account))
 	if err != nil {
 		return 0, err
@@ -910,7 +920,18 @@ func (s *service) getBalance(account accountName) (uint64, error) {
 
 // this only supports deriving segwit v0 accounts
 func (s *service) deriveNextAddress(account accountName) (btcutil.Address, error) {
+	if err := s.expectWallet(); err != nil {
+		return nil, err
+	}
+
 	return s.wallet.NewAddress(lnwallet.WitnessPubKey, false, string(account))
+}
+
+func (s *service) expectWallet() error {
+	if s.wallet == nil {
+		return fmt.Errorf("wallet not initialized")
+	}
+	return nil
 }
 
 func withChainSource(chainSource chain.Interface) WalletOption {

--- a/server/internal/infrastructure/wallet/btc-embedded/wallet.go
+++ b/server/internal/infrastructure/wallet/btc-embedded/wallet.go
@@ -68,9 +68,10 @@ const (
 )
 
 var (
-	p2wpkhKeyScope     = waddrmgr.KeyScopeBIP0084
-	p2trKeyScope       = waddrmgr.KeyScopeBIP0086
-	outputLockDuration = time.Minute
+	ErrWalletNotInitialized = fmt.Errorf("wallet not initialized")
+	p2wpkhKeyScope          = waddrmgr.KeyScopeBIP0084
+	p2trKeyScope            = waddrmgr.KeyScopeBIP0086
+	outputLockDuration      = time.Minute
 )
 
 type service struct {
@@ -221,7 +222,7 @@ func NewService(cfg WalletConfig, options ...WalletOption) (ports.WalletService,
 }
 
 func (s *service) Close() {
-	if s.wallet != nil {
+	if s.isInitialized() {
 		if err := s.wallet.Stop(); err != nil {
 			log.WithError(err).Warn("failed to gracefully stop the wallet, forcing shutdown")
 		}
@@ -245,7 +246,7 @@ func (s *service) Restore(_ context.Context, seed, password string) error {
 }
 
 func (s *service) Unlock(_ context.Context, password string) error {
-	if s.wallet == nil {
+	if !s.isInitialized() {
 		pwd := []byte(password)
 		opt := btcwallet.LoaderWithLocalWalletDB(s.cfg.Datadir, false, time.Minute)
 		config := btcwallet.Config{
@@ -325,8 +326,8 @@ func (s *service) Unlock(_ context.Context, password string) error {
 }
 
 func (s *service) Lock(_ context.Context, _ string) error {
-	if err := s.expectWallet(); err != nil {
-		return err
+	if !s.isInitialized() {
+		return ErrWalletNotInitialized
 	}
 
 	s.wallet.InternalWallet().Lock()
@@ -583,7 +584,7 @@ func (s *service) SignTransactionTapscript(ctx context.Context, partialTx string
 }
 
 func (s *service) Status(ctx context.Context) (ports.WalletStatus, error) {
-	if s.wallet == nil {
+	if !s.isInitialized() {
 		return status{}, nil
 	}
 
@@ -906,8 +907,8 @@ func (s *service) initWallet(wallet *btcwallet.BtcWallet) error {
 }
 
 func (s *service) getBalance(account accountName) (uint64, error) {
-	if err := s.expectWallet(); err != nil {
-		return 0, err
+	if !s.isInitialized() {
+		return 0, ErrWalletNotInitialized
 	}
 
 	balance, err := s.wallet.ConfirmedBalance(0, string(account))
@@ -920,18 +921,15 @@ func (s *service) getBalance(account accountName) (uint64, error) {
 
 // this only supports deriving segwit v0 accounts
 func (s *service) deriveNextAddress(account accountName) (btcutil.Address, error) {
-	if err := s.expectWallet(); err != nil {
-		return nil, err
+	if !s.isInitialized() {
+		return nil, ErrWalletNotInitialized
 	}
 
 	return s.wallet.NewAddress(lnwallet.WitnessPubKey, false, string(account))
 }
 
-func (s *service) expectWallet() error {
-	if s.wallet == nil {
-		return fmt.Errorf("wallet not initialized")
-	}
-	return nil
+func (s *service) isInitialized() bool {
+	return s.wallet != nil
 }
 
 func withChainSource(chainSource chain.Interface) WalletOption {

